app/src/main/java/com/babylon/wallet/android/domain/usecases/ResolveAccountsLedgerStateUseCase.kt

```diff
@@ -0,0 +1,86 @@
+package com.babylon.wallet.android.domain.usecases
+
+import com.babylon.wallet.android.data.gateway.apis.StateApi
+import com.babylon.wallet.android.data.gateway.extensions.defaultDepositRule
+import com.babylon.wallet.android.data.gateway.extensions.isEntityActive
+import com.babylon.wallet.android.data.gateway.extensions.toProfileDepositRule
+import com.babylon.wallet.android.data.gateway.generated.models.DefaultDepositRule
+import com.babylon.wallet.android.data.gateway.generated.models.StateEntityDetailsOptIns
+import com.babylon.wallet.android.data.gateway.generated.models.StateEntityDetailsRequest
+import com.babylon.wallet.android.data.gateway.model.ExplicitMetadataKey
+import com.babylon.wallet.android.data.repository.toResult
+import com.babylon.wallet.android.di.JsonConverterFactory
+import com.babylon.wallet.android.di.ShortTimeoutGatewayHttpClient
+import com.babylon.wallet.android.di.buildApi
+import com.babylon.wallet.android.domain.model.AccountWithOnLedgerStatus
+import okhttp3.OkHttpClient
+import rdx.works.profile.data.model.apppreferences.Radix
+import rdx.works.profile.data.model.currentGateway
+import rdx.works.profile.data.model.pernetwork.Network
+import rdx.works.profile.data.repository.ProfileRepository
+import retrofit2.Converter
+import javax.inject.Inject
+
+class ResolveAccountsLedgerStateUseCase @Inject constructor(
```

Comment:
ResolveAccountWithOnLedgerStateUseCase ?

Reply:
yup, this is not a use case but a network client, or repository the way we call it here.
ResolveAccountsLedgerStateRepository and better to be in the data package